apache/couchdb-fauxton
# WIP - [partitioned dbs] Show results for partitioned views and All Docs filtered by partition key

## Overview

**!!!!! Depends on #1148 being merged first !!!!!!**

Fifth in a series of PRs that will be submitted in support of the new user-defined partitioned databases feature (apache/couchdb#1605).

Summary:

 - Show results for partitioned views
 - Filter All Documents by partition key (when one is selected)
 - Show warning message when a global view is selected and a partition key is selected
 - Show warning message when a partitioned view is selected but no key is selected

## Testing recommendations

Scenario 1
- Go to All Documents
- In the Partition Selector, type in a key then press Enter
  - Verify the list of docs is updated to show only docs that match the partition key
- Choose a different partition key
  - Verify the list of docs is updated again
- Click the Filter icon in the Partition Selector to switch back to Global mode
  - Verify list of docs is updated

Scenario 2
- Create a new view `p1` in a new ddoc called `ddocPartitioned`
  _- NOTE: at the moment, the Index Editor doesn't preserve the selected partition key. It'll be updated in a future PR_
- Type in a partition key then press Enter
  - Verify the view results is now displayed correctly
  - Verify the JSON and Metadata buttons are not visible
- Click the Filter icon in the Partition Selector to switch back to Global mode
  - Verify a warning is displayed

Scenario 3
- Create a new view `g1` in a new ddoc called `ddocGlobal`
  - edit `ddocGlobal` and add `"options": { "partitioned": false }`
  - verify that `ddocPartitioned` and `ddocGlobal` have different icons on the sidebar
- Click the `g1` view
  - verify the results are displayed
- Type in a partition key then press Enter
  - verify a warning is displayed

## Related Pull Requests

apache/couchdb#1605

## Checklist
